pkg/account/domain/account.go

```diff
@@ -106,3 +113,61 @@ func (a *AccountV2) Disable() error {
 	a.UpdatedAt = time.Now().Unix()
 	return nil
 }
+
+func (a *AccountV2) AddSearchFilter(
+	name string,
+	query string,
+	targetType proto.FilterTargetType,
+	environmentId string, defaultFilter bool) error {
+	id, err := uuid.NewUUID()
+	if err != nil {
+		return err
+	}
+
+	searchFilter := &proto.SearchFilter{
+		Id:               id.String(),
+		Name:             name,
+		Query:            query,
+		FilterTargetType: targetType,
+		EnvironmentId:    environmentId,
+		DefaultFilter:    defaultFilter,
+	}
+	a.AccountV2.SearchFilters = append(a.AccountV2.SearchFilters, searchFilter)
+	a.UpdatedAt = time.Now().Unix()
+	return nil
+}
+
+func (a *AccountV2) DeleteSearchFilter(id string) error {
+	var searchFilters []*proto.SearchFilter
+	var isFound = false
+	for i, f := range a.AccountV2.SearchFilters {
+		if f.Id == id {
+			searchFilters = append(a.AccountV2.SearchFilters[:i], a.AccountV2.SearchFilters[i+1:]...)
+			isFound = true
+			continue
+		}
+	}
+	if len(searchFilters) == 0 {
+		if isFound {
+			a.AccountV2.SearchFilters = nil
+			a.UpdatedAt = time.Now().Unix()
+			return nil
+		} else {
+			return errSearchFilterNotFound
+		}
+	}
+	a.AccountV2.SearchFilters = searchFilters
+	a.UpdatedAt = time.Now().Unix()
+	return nil
```

Comment:
```suggestion
	if !isFound {
	    return errSearchFilterNotFound
	}
	a.AccountV2.SearchFilters = searchFilters
	a.UpdatedAt = time.Now().Unix()
	return nil
```

Reply:
I Changed to delete the filter and exit the function when the ID is found without using `isFound`.
https://github.com/bucketeer-io/bucketeer/blob/feature-search-filter-proto-domain/pkg/account/domain/account.go#L140